### DISCUSSION TITLE
Inquiry about Installing Only DatePicker Components in Rsuite

### DISCUSSION THREAD

**Discussion by alokVerma749** [rsuite/rsuite#3553] [General]
I am currently working on a project where I need to integrate a Datepicker component.

After researching various options, I found Rsuite and was impressed by its comprehensive suite of React components. However, for my specific project requirements, I only need the Calendar/Time/Date Picker components from Rsuite.

I was wondering if there is a way to install only these specific components without including the entire Rsuite package. This would help streamline my project and reduce unnecessary dependencies.

I'm looking forward to your guidance on this matter.

**Comment by simonguo** [comment]
Hi @alokVerma749,  you only import DatePicker  and Calendar 

```js
import DatePicker from 'rsuite/DatePicker';
import Calendar from 'rsuite/Calendar';
```

CSS import `rsuite-no-reset.min.css`, will not affect the style in your project.
```js
import 'rsuite/dist/rsuite-no-reset.min.css';
```

**Reply by alokVerma749** [reply]
Actually I want only `DateRangePicker`, `DateTimeRangePicker` and  `DatePicker` component, is there any way i can only download these three component and not whole package??

**Reply by simonguo** [reply]
```tsx
import DatePicker from 'rsuite/DatePicker';
import DateRangePicker from 'rsuite/DateRangePicker';
```
There is no `DateTimeRangePicker` component, you only need to control whether to display the time through the `format` prop

**Reply by alokVerma749** [reply]
@simonguo yes I got it, but the problem is the unpack size of Rsuite package is 20MB which is expensive for us, So can you please tell me whether can I install only certain components (`DatePicker`, `DateRangePicker`) from the library instead of whole library Rsuite?

**Reply by simonguo** [reply]
Currently, other components except Table do not support independent install

**Comment by SevenOutman** [comment]
Hi @alokVerma749, if you're using a bundler like webpack to build your project, it should already be excluding unused modules from the output artifacts. So it shouldn't be a problem to install the complete library.
